app/src/main/java/com/lagradost/cloudstream3/ui/settings/SettingsUI.kt

```diff
@@ -187,5 +194,12 @@ class SettingsUI : PreferenceFragmentCompat() {
             return@setOnPreferenceClickListener true
         }
 
+        setKey( // default value is layout specific
```

Comment:
This is just wrong, as setKey will only set the first time you go into settings. If you want this behavior, then dont use a SwitchPreference, and instead use a bottom popup with the options `show`, `dont show`, and `auto` like we do App Layout.

Reply:
A two state button cant capture a tri state system.

Reply:
I don't see this being a tri-state system, there is only `show` or `dont_show`. Only extra factor here being the default value of the switch only valuable when the `key` isnt in the sharedpref yet. `setKey` works in that case, only there because the default value cannot be set in the xml since it needs some logic. I agree on if there were to be more than 2 states, the sheet would have been a better idea, switch should work for us here.

Reply:
It *is* a tri-state system

### Clean install: 
No value is set for `confirm_exit_key`

```kt
val confirmBeforeExit = settingsManager.getBoolean(getString(R.string.confirm_exit_key), SettingsUI.getConfirmExitDialogDefault())
```
As no value is set it will default to:
```kt
SettingsUI.getConfirmExitDialogDefault()) 
```
Which equals to `isLayout(TV or EMULATOR)`, **which is dynamic and based on the current layout.**

### Visited settings once:
The setting behavior is **pinned** based on the current layout used when visiting settings.
This is done when doing 
```kt
setKey(
       getString(R.string.confirm_exit_key),
            settingsManager.getBoolean(
                getString(R.string.confirm_exit_key),
                getConfirmExitDialogDefault()
            )
        )
```
Which essentially evaluates to `setKey(getString(R.string.confirm_exit_key), isLayout(TV or EMULATOR))` on a clean install.

### Changed the setting manually:
The value is also pinned to another value, not based on layout.

--- 
So we have:
1. Based on layout
2. True
3. False

Whereas if you go to either (2) true or (3) false it is not possible to restore it to (3) dynamic layout.

To demonstrate the issue with this system we can imagine a off-brand Android TV which is not automatically recognized as a TV. In this case when the used goes into layout settings to change their UI this value will be set to **false**. This means that the user, while never doing anything related to this setting will have it permanently turned off, without even knowing it existed.

You **need** a tri-state option if you want the dialog to be able to be shown in phones, as opposed to the old behavior of only TVs.

Reply:
I believe this makes sense. Thank you for the detailed analysis, will fix and get back!